app/models/middleware_server.rb

```diff
@@ -5,4 +5,19 @@ class MiddlewareServer < ActiveRecord::Base
   has_many :middleware_deployments, :foreign_key => "server_id", :dependent => :destroy
   serialize :properties
   acts_as_miq_taggable
+
+  # Callback from _single_quad.html.haml to determine the icon
+  # we want to display a different icon depending of the type
+  # of server we have.
+  # See also MiddlewareServerController#listicon_image for the counterpart for lists.
+  def item_image
```

Comment:
@matthewd I knew the icon does not belong in a model, but forgot where it does belong.  :-)
@pilhuhn you can find existing decorators in `app/decorators` directory.  Please let one of us know if you need any help.